source/UI.cpp

```diff
@@ -28,6 +28,9 @@ using namespace std;
 // of them handles it. If none do, this returns false.
 bool UI::Handle(const SDL_Event &event)
 {
+	// Handle any queued push or pop commands.
+	PushOrPop();
+
 	bool handled = false;
```

Comment:
No. That's the cleanest version I can think of to fix one problem with the integration tests. The problem is:

1. When `LoadingPanel` is done loading the game data, it pops itself and pushes a new `MenuPanel`.
2. Later, we reach `Test::Step`, which executes the first step of the test to run (since the data is loaded).
3. In the next frame, SDL events are processed and sent to the menu panels.
4. But the menu panels are empty! The `MenuPanel` that is added only gets actually pushed inside the `StepAll` (or after handling any input processing).
5. So the first input step of the Test is lost.